### PR TITLE
Fixes #28232 - ensure validator for IPAM lists all available ipam types

### PR DESCRIPTION
**Pull request by evgeni**
also do the same for the bootmode

### CODE REVIEW

File: app/controllers/api/v2/subnets_controller.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def show
           param :gateway, String, :desc => N_("Subnet gateway")
           param :dns_primary, String, :desc => N_("Primary DNS for this subnet")
           param :dns_secondary, String, :desc => N_("Secondary DNS for this subnet")
-          param :ipam, String, :desc => N_('IP Address auto suggestion mode for this subnet, valid values are "DHCP", "Internal DB", "None"')
+          param :ipam, IPAM::MODES.values, :desc => N_('IP Address auto suggestion mode for this subnet.')
```
Comment (tbrisker) [inline app/controllers/api/v2/subnets_controller.rb:43]: Should we still list all the valid modes? e.g something like:
```suggestion
          param :ipam, IPAM::MODES.values, :desc => (N_("IP Address auto suggestion mode for this subnet. Valid values are: %s") %  IPAM::MODES.values.join(','))
```
Or is this something apipie does already when using a list validator?
```

Comment (evgeni) [reply app/controllers/api/v2/subnets_controller.rb:43]: That's how it looks after my patch:

Screenshot from 2019-11-11 11-33-19

Comment (evgeni) [reply app/controllers/api/v2/subnets_controller.rb:43]: Also, making the description dynamic makes it hard for translations :)

**Review by tbrisker — APPROVED** [review]
Thanks @evgeni !

**Comment by tbrisker** [comment]
1.24 - f09ad5a05

**Pull request merged**